src/main/java/com/askimed/nf/test/util/FileUtil.java

```diff
@@ -105,7 +105,21 @@ public static String makeRelative(File baseDir, File absoluteFile) {
 	public static String getMd5(Path self) throws IOException, NoSuchAlgorithmException {
 		Formatter fm = new Formatter();
 		MessageDigest md = MessageDigest.getInstance("MD5");
-		md.update(Files.readAllBytes(self));
+		// for .gz files, calculate md5 hash on decompressed content
+		if (self.toString().endsWith(".gz")) {
+			FileInputStream fis = new FileInputStream(self.toString());
+			GZIPInputStream gzis = new GZIPInputStream(fis);
+			byte[] buffer = new byte[4096];
+			int read = gzis.read(buffer);
+			while ( read >= 0) {
+				md.update(buffer, 0, read);
+				read = gzis.read(buffer);
+			}
+		// for other files, calculate md5 hash directly from file

```

Comment:
`gzis.close()` is missing. The rest looks good 👍  Thanks for you help!

Reply:
Added this! Also closed `fis`. I do not how to code in Java but am learning by doing ;-)